rust_src/src/editfns.rs

```diff
@@ -182,3 +182,43 @@ pub fn char_after(mut pos: LispObject) -> LispObject {
         }
     }
 }
+
+/// Return a copy of STRING with text properties added.
+/// First argument is the string to copy.
+/// Remaining arguments form a sequence of PROPERTY VALUE pairs for text
+/// properties to add to the result.
+/// usage: (propertize STRING &rest PROPERTIES)  */
+#[lisp_fn(min = "1")]
+pub fn propertize(args: &mut [LispObject]) -> LispObject {
+    /* Number of args must be odd. */
+    if args.len() & 1 == 0 {
+        error!("Wrong number of arguments");
+    }
+
+    let mut it = args.iter();
+
+    let first = it.next().unwrap(); // safe, there is at least 1 arg
+    if !first.is_string() {
+        wrong_type!(Qstringp, *first);
+    }
+
+    let copy = first.clone();
```

Comment:
Unfortunately, this is not doing what you would expect. You are calling the 'clone' method of `LispObject`. `LispObject`'s are just tagged pointers, so this is really just a pointer copy. You will want to copy the C layer's call to Fcopy_sequence.

Reply:
Ah, I had used Fcopy_sequence and then thought I was doing the right thing by using the Rust method. Easy fix. Thanks.